### PR TITLE
Added tokens to RESTful service endpoints

### LINKED ISSUE #131: Add admin token protection to necessary endpoints

### PR DESCRIPTION
**Pull request by T-Syme**
If possible, please check to make sure that the correct users are being validated in each case

### CODE REVIEW

File: RepairShop-Backend/src/main/java/ca/mcgill/ecse321/repairshop/controller/AppointmentController.java [hunk 1]
```diff
@@ -14,15 +15,22 @@
...
     /** Endpoint to create an appointment
      * @param startTimestamp Timestamp for the start time of the appointment as a string
      * @param serviceName The name of the service for the appointment
      * @param customerEmail The email of the customer for whom to book the appointment
+     * @param token for the admin or customer to create an appointment
      * @return the appointment that's been created
      */
     @PostMapping("/create")
-    public ResponseEntity<?> createAppointment(@RequestParam String startTimestamp, @RequestParam String serviceName, @RequestParam String customerEmail) {
+    public ResponseEntity<?> createAppointment(@RequestParam String startTimestamp, @RequestParam String serviceName, @RequestParam String customerEmail, @RequestHeader String token) {
```
Comment (JDziewonski98) [inline RepairShop-Backend/src/main/java/ca/mcgill/ecse321/repairshop/controller/AppointmentController.java]: I think it is much better to use request body for POST requests

Comment (JDziewonski98) [reply RepairShop-Backend/src/main/java/ca/mcgill/ecse321/repairshop/controller/AppointmentController.java]: usually request params are to identify a resource uniquely per REST conventions, while body is to pass general data

File: RepairShop-Backend/src/main/java/ca/mcgill/ecse321/repairshop/controller/BusinessController.java [hunk 2]
```diff
@@ -80,13 +93,19 @@
...
+    /**
      * Add a new holiday to the business.
+     * @param startDateTime start time of the holiday
+     * @param endDateTime end time of the holiday
+     * @param token of the admin
      * @return a list of all holidays
      */
     @PostMapping("/create/holidays")
-    public ResponseEntity<?> addHoliday(@RequestParam Timestamp startDateTime, @RequestParam Timestamp endDateTime) {
+    public ResponseEntity<?> addHoliday(@RequestParam Timestamp startDateTime, @RequestParam Timestamp endDateTime, @RequestParam String token) {
```
Comment (JDziewonski98) [inline RepairShop-Backend/src/main/java/ca/mcgill/ecse321/repairshop/controller/BusinessController.java]: should use body

File: RepairShop-Backend/src/main/java/ca/mcgill/ecse321/repairshop/controller/BusinessController.java [hunk 3]
```diff
@@ -95,14 +114,19 @@
...
      * Delete a holiday from the business.
+     * @param startDateTime start time of the holiday
+     * @param endDateTime end time of the holiday
+     * @param token of the admin
      * @return a list of all holidays
      */
     @GetMapping("/delete/holidays")
-    public ResponseEntity<?> deleteHoliday(@RequestParam Timestamp startDateTime,
-                                        @RequestParam Timestamp endDateTime) {
+    public ResponseEntity<?> deleteHoliday(@RequestParam Timestamp startDateTime, @RequestParam Timestamp endDateTime, @RequestParam String token) {
```
Comment (JDziewonski98) [inline RepairShop-Backend/src/main/java/ca/mcgill/ecse321/repairshop/controller/BusinessController.java]: maybe a bit late but could take id of a holiday? You call get all holidays, choose an id, and pass it to this

File: RepairShop-Backend/src/main/java/ca/mcgill/ecse321/repairshop/controller/CustomerController.java [hunk 4]
```diff
@@ -12,137 +15,166 @@
...
+    /**
+     * POST request to change a password for a customer
+     *
+     * @param email       of customer
+     * @param newPassword of customer
+     * @param token for the customer to change password
+     * @return a customer dto
+     */
+    @PostMapping("/changePassword/{email}")
+    public ResponseEntity<?> changePassword(@PathVariable("email") String email, @RequestParam String newPassword, @RequestHeader String token) {
```
Comment (JDziewonski98) [inline RepairShop-Backend/src/main/java/ca/mcgill/ecse321/repairshop/controller/CustomerController.java]: should definitely use body for new password

File: RepairShop-Backend/src/main/java/ca/mcgill/ecse321/repairshop/controller/ReminderController.java [hunk 5]
```diff
@@ -37,12 +45,16 @@
      * @param serviceName         name of the service
      * @param type                of the new reminder
      * @param email               of the customer associated to the new reminder
+     * @param token               of the admin
      * @return the new reminder if created successfully
      */
     @PostMapping("/create")
     public ResponseEntity<?> createReminder(@RequestParam String dateTime, @RequestParam String appointmentDateTime,
-                                            @RequestParam String serviceName, @RequestParam String type, @RequestParam String email) {
+                                            @RequestParam String serviceName, @RequestParam String type, @RequestParam String email, @RequestHeader String token) {
```
Comment (JDziewonski98) [inline RepairShop-Backend/src/main/java/ca/mcgill/ecse321/repairshop/controller/ReminderController.java]: body here

File: RepairShop-Backend/src/main/java/ca/mcgill/ecse321/repairshop/controller/ServiceController.java [hunk 6]
```diff
@@ -45,11 +49,15 @@
      * @param name of the new service
      * @param duration of the new service
      * @param price of the new service
+     * @param token of the admin
      * @return the new service if created successfully
      */
     @PostMapping("/create/{name}")
-    public ResponseEntity<?> createService(@PathVariable String name, @RequestParam int duration, @RequestParam double price) {
+    public ResponseEntity<?> createService(@PathVariable String name, @RequestParam int duration, @RequestParam double price, @RequestHeader String token) {
```
Comment (JDziewonski98) [inline RepairShop-Backend/src/main/java/ca/mcgill/ecse321/repairshop/controller/ServiceController.java]: body here

File: RepairShop-Backend/src/main/java/ca/mcgill/ecse321/repairshop/controller/TechnicianController.java [hunk 7]
```diff
@@ -53,13 +58,18 @@
 	 * POST request to change a password for a technician
 	 * @param email of technician
 	 * @param newPassword of technician
+	 * @param token of logged in technician making the request
 	 * @return a technician dto
 	 */
 	@PostMapping("/changePassword/{email}")
-	public ResponseEntity<?> changePassword(@PathVariable("email") String email, @RequestParam String newPassword){
+	public ResponseEntity<?> changePassword(@PathVariable("email") String email, @RequestParam String newPassword, @RequestHeader String token){
```
Comment (JDziewonski98) [inline RepairShop-Backend/src/main/java/ca/mcgill/ecse321/repairshop/controller/TechnicianController.java]: body for password

**Review by JDziewonski98 — APPROVED** [review]
made some comments about where in my opinion a different type of request input should be used, good to merge other than that tho

**Review by Shaswata79 — APPROVED** [review]

**Pull request merged**